### PR TITLE
Add a production test build to CI Tests

### PR DESCRIPTION
**Pull request by eatyourgreens**
Set up a copy of the production build environment. Run a full install (to install the build tools.) Run a test build of all the libraries and the NextJS apps. Run the tests with PANOPTES_ENV=test.

#1943 restricts the Jenkins builds to deploys only, so this runs a full build of the monorepo as part of the CI tests for each PR. In theory, this should catch breaking changes in the NextJS apps before we merge to staging.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/app-content-pages/server/set-logging.spec.js [hunk 1]
```diff
@@ -10,7 +10,13 @@ describe('Server > setLogging', function () {
...
   describe('default behaviour', function () {
     before(function () {
-      setLogging = require('./set-logging')
+      setLogging = proxyquire('./set-logging', {
+        process: {
+          env: {
+            NODE_ENV: 'development'
+          }
+        }
+      })
```
Comment (srallen) [inline packages/app-content-pages/server/set-logging.spec.js:19]: is the descriptor still accurate for this if you have to configure the node env? Is this still default behavior?

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -10,7 +10,13 @@ jobs:
...
    runs-on: ubuntu-latest
    env:
-     CI: true
+     APP_ENV: production
+     NODE_ENV: production
+     PANOPTES_ENV: production
+     CONTENTFUL_ACCESS_TOKEN: ${{ secrets.CONTENTFUL_ACCESS_TOKEN }}
+     CONTENTFUL_SPACE_ID: ${{ secrets.CONTENTFUL_SPACE_ID }}
+     CONTENT_ASSET_PREFIX: 'https://fe-content-pages.zooniverse.org'
+     PROJECT_ASSET_PREFIX: 'https://fe-project.zooniverse.org'
```
Comment (srallen) [inline .github/workflows/ci-tests.yml:19]: Could you add the missing environment variables and their definitions to https://github.com/zooniverse/front-end-monorepo/blob/master/README.md#environment-variables

**Review by srallen — APPROVED** [review]
I think this is fine. I have a non-blocking question about the change in the set-logging spec file. In addition, I think it'd be really helpful to add a section to the repo readme about testing, both CI and manual options.

File: packages/app-content-pages/server/set-logging.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-content-pages/server/set-logging.spec.js:19]: I think this is the default, based on a very quick look at `set-logging.js`. I'm not sure. The default webpack mode might be 'none'. https://webpack.js.org/configuration/mode/

File: .github/workflows/ci-tests.yml (hunk 2 above)
Comment (eatyourgreens) [reply .github/workflows/ci-tests.yml:19]: Sure. I'll update this PR.

**Comment by eatyourgreens** [comment]
I thought we had a `Contributing.md` file with instructions for running the tests, but maybe not.

File: packages/app-content-pages/server/set-logging.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-content-pages/server/set-logging.spec.js:19]: I've changed the wording on these, because I'm not 100% sure what 'default' means here.

Comment (eatyourgreens) [reply packages/app-content-pages/server/set-logging.spec.js:19]: The original test assumed that the default `NODE_ENV` was 'development' or undefined. Now that the tests are being run against a production build, the default `NODE_ENV` is 'production'.

**Pull request merged**